### PR TITLE
helper Subject.isUserInSubject

### PR DESCRIPTION
**Pull request by Nauna**
if the user is creating a business, the business is not in the context yet, thus does not have permission to give another user a ucj... however if you set the business into the context, then all is good.
^^ happening https://github.com/nanoPayinc/NANOPAY/pull/13391/files#diff-62036d96ea096f4b33e16a5ad67815616607ae2f4ab7ca1448ac9871e67f1339R56

I'm adding a helper function to subject cause it could be useful in other places.

### CODE REVIEW

**Review by smirnovaka — APPROVED** [review]

File: src/foam/nanos/auth/Subject.js [hunk 1]
```diff
@@ -68,6 +68,23 @@ foam.CLASS({
...
+        'Long idCheck'
+      ],
+      javaCode: `
+      for ( User user : getUserPath() ) {
+        if ( foam.util.SafetyUtil.equals(user.getId(), idCheck) ) return true;
+      }
+      return false;
+      `,
+      code: function() {
+        return !! this.userPath.filter(u => u.id == idCheck);
```
Comment (kgrgreer) [inline src/foam/nanos/auth/Subject.js]: I think find() would be better than filter().

**Comment by jlhughes** [comment]
Regarding naming, thoughts on 
has(userId) ?

File: src/foam/nanos/auth/Subject.js [hunk 2]
```diff
@@ -68,6 +68,23 @@ foam.CLASS({
   ],
 
   methods: [
+    {
+      name: 'isUserInSubject',
```
Comment (kgrgreer) [inline src/foam/nanos/auth/Subject.js:72]: Name seems good.

File: src/foam/nanos/auth/Subject.js (hunk 1 above)
Comment (Nauna) [inline src/foam/nanos/auth/Subject.js]:
```suggestion
        return !! this.userPath.find(u => u.id == idCheck);
```

**Pull request merged**